src/ir/value.cc

```diff
@@ -0,0 +1,43 @@
+//-----------------------------------------------------------------------------
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//----------------------------------------------------------------------------
+#include "src/ir/value.h"
+
+#include <vector>
+
+#include "absl/strings/str_format.h"
+#include "src/common/utils/overloaded.h"
+#include "src/ir/ssa_names.h"
+
+namespace raksha::ir {
+
+std::string Value::ToString(SsaNames& ssa_names) const {
+  return std::visit(
```

Comment:
Didn't we decide that this change is not needed?

Reply:
I realized later that the change we discussed is not possible. I think what we agreed to do was something like this.   
`
std::string ToString(SsaNames& ssa_names) const {
    return absl::StrFormat("%s", ssa_names.GetOrCreateID(Value(*element_));
  }
`
In place of https://github.com/google-research/raksha/blob/c8c4593d6febcbea31ef864ca3e2267bac5d38e1/src/ir/value.h#L52

We would need to know the variant of value which is not something we can do in NameValue class. Let me know if you had other thoughts on this.

Reply:
Moving Value changes and discussions to another PR.